### PR TITLE
Added prn prescription

### LINKED ISSUE #3194: PRN Prescription
We have Prescription Section in the Consultation form (https://care.coronasafe.in/facility/015ad911-39b7-4104-b304-bf8178c33830/patient/76742faa-30dd-4eb9-9649-fe718f947ec0/consultation/2ad89b7c-7bff-4035-a3c6-36d53551a8f0/update) 

Right below the prescription section, a new section for "PRN Prescription" needs to be added.

**Info:** A PRN- Pro Re Nata Prescription is used when the doctor needs to prescribe medicines which is not scheduled, but that needs to be administered upon occurrence or an event/any given circumstance.
eg: The doctor might instruct the nurse to give a High dose of Paracetamol to a patient if the temp. goes above X degree. 

**Requirement:** 
Screenshot 2022-07-22 at 4 53 42 PM
The design is only suggestive. Keep the design consistent with the prescription section.

The PRN Prescription is to have
1. Medicine name- same as the prescription module.
2. Route-  same as the prescription module
3. Dosage- same as the prescription module
4. Indicator- this must be a text field to accommodate 100 characters. (this is where the doctor will specify the circumstance under which the medicine should be given, eg: temp goes above X degrees, BP goes below X )
5. Max. Dosage in 24 hrs: Input to be same as the dosage
6. Min time between 2 doses: Input field to specify __ Hrs . user to enter No. with dropdown to show [1,2,3,6,12,24]

- The PRN Prescription must appear right below the prescription
- There must be option to add more than 1 PRN medicine with all the above options

### LINKED ISSUE #3301: Allow decimals of dosage
The dosage in the prescription module needs to allow input of decimals of up to 2 places. 

This holds good for prescription and pnr prescripton

### PR DESCRIPTION
**Pull request by skks1212**
closes #3194 
closes #3301 
image

### CODE REVIEW

File: src/Components/Common/prescription-builder/PRNPrescriptionBuilder.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    route: "",
+    dosage: "0 mg",
+    max_dosage: "0 mg",
+    indicator: "",
+    min_time: 0
+}
+
+const dosageHrs = [1,2,3,6,12,24];
+
+export default function PRNPrescriptionBuilder(props : {prescriptions :PRNPrescriptionType[], setPrescriptions : React.Dispatch<React.SetStateAction<PRNPrescriptionType[]>>}){
```
Comment (developedBySJ) [inline src/Components/Common/prescription-builder/PRNPrescriptionBuilder.tsx]: Extract `props` interface

File: src/Components/Common/prescription-builder/PRNPrescriptionBuilder.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+export const PRNEmptyValues = {
+    medicine: "",
+    route: "",
+    dosage: "0 mg",
+    max_dosage: "0 mg",
+    indicator: "",
+    min_time: 0
+}
+
+const dosageHrs = [1,2,3,6,12,24];
```
Comment (developedBySJ) [inline src/Components/Common/prescription-builder/PRNPrescriptionBuilder.tsx]:
```suggestion
const DOSAGE_HRS= [1,2,3,6,12,24];
```

**Review by developedBySJ — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@skks1212 

image

- [ ] dosage more than 2 decimals are been accepted, but in issue, it says up to 2 decimals only

image

- [ ] when we are clicking on the increase and decrease button, the decimal inputted are been disappearing

**Comment by shivankacker** [comment]
@nihal467 I have not added any restriction just in case anyone would like to use any more decimals. Regarding number 2, that is a browser specific functionality, so it will always round it up to nearest 1s.

**Comment by nihal467** [comment]
test approved

File: src/Components/Facility/ConsultationForm.tsx [hunk 3]
```diff
@@ -235,14 +239,9 @@ export const ConsultationForm = (props: any) => {
...
-      if (
-        res &&
-        res.data &&
-        res.data.discharge_advice &&
-        Object.keys(res.data.discharge_advice).length != 0
-      ) {
-        setDischargeAdvice(res && res.data && res.data.discharge_advice);
-      }
+      setDischargeAdvice(res && res.data && res.data.discharge_advice);
+      console.log("recieved", res.data.prn_prescription);
```
Comment (khavinshankar) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
```

**Review by khavinshankar — APPROVED** [review]

**Comment by khavinshankar** [comment]
@developedBySJ can you please rereview this one?

**Comment by nihal467** [comment]
test approved

**Review by gigincg — APPROVED** [review]

**Pull request merged**